### PR TITLE
DS-808 Profile edit form mockup

### PR DESCRIPTION
**Pull request by MarcinMr**
## Jira

https://pegadigitalit.atlassian.net/browse/DS-808

## Summary

A mockup page was created in the User folder -> `/pattern-lab/?p=pages-user-profile-edit`

## Details

A Profile Edit Form was created similar to the design attached in the  [Jira task](https://pegadigitalit.atlassian.net/browse/SN-243). The card's design is omitted, instead, a simple page with form inputs is created.

## How to test

Pull the branch. Make sure that the form is created in a proper way and all necessary inputs are included.

### CODE REVIEW

**Review by mikemai2awesome — CHANGES_REQUESTED** [review]
@MarcinMr Good job with this. Please fix the following issues:

1. Add a page heading with `h1`. It can be "Edit Profile".
1. Heading orders: each section should be an `h2`, subsequent headings in each section should be `h3`.

    Screen Shot 2022-06-24 at 12 20 39 PM

1. Fix missing labels for certain inputs.

    Screen Shot 2022-06-24 at 12 21 44 PM

You can use either Sa11y bookmarklet or Chrome WAVE extension to detect these issues.

**Comment by MarcinMr** [comment]
@mikemai2awesome 
- H1 is added and the hierarchy of the headings is fixed. 
- Fixed input labels
- Moved page from the sandbox to the User folder

**Review by mikemai2awesome — APPROVED** [review]
All set from me.

**Comment by krlucas** [comment]
@MarcinMr @mikemai2awesome 
This looks great. A couple of requests/comments.

* Can we up with examples of two or more existing
** Projects
*** With multiple products already added
** Languages
** Project Roles
** Pega Specializations
* Should Project > Product, Roles and Specializations use the React multi-select? cc @danielamorse

File: docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig [hunk 1]
```diff
@@ -0,0 +1,1535 @@
...
+      {% endset %}
+
+      {% set phone_number %}
+        {% set label %}
+          {% include '@bolt-components-form/form-label.twig' with {
+            title: 'Phone Number (required)',
+            displayType: 'block',
+            attributes: {
+              for: 'phone-number'
+            }
```
Comment (ajfwebdev) [inline docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig:127]: @MarcinMr Based on the image of the mockup, we'll need some changes for phone number editing. It should work like it does today on the user profile edit form on MyPega. Pega users like us are not allowed to edit the phone number, so we can only see the current phone number string (including country dial code) and the extension, if any. If you login to QA as a non-pega user, you will see a button to click if you wish to edit the phone number, and new fields for editing will appear. There will be one select field for selecting the country dial code, one field for typing in the rest of the phone number, and a third for typing in the extension.

Comment (krlucas) [reply docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig:127]: This is how the phone number works in MyPega


Then click the pencil Icon


Search for a country

Comment (MarcinMr) [reply docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig:203]: @ajfwebdev @krlucas 

I've made some changes here related to the editing phone number fieldset. Is it what we are looking for?

**Comment by krlucas** [comment]
@MarcinMr I'll defer to @ajfwebdev and Tomasz on the phone number. It looks workable to me.

I'd still like you update with an example of a user who has existing Projects, Languages, Roles and Pega Specializations.

Here's a screenshot of what that looks like in MyPega

**Comment by MarcinMr** [comment]
@krlucas thanks for the screenshot. I updated the mockup with the existing user projects and selected languages. I cannot update the `Project Roles` and `Pega Specializations` like on the screen. In this case, the multiselect is the solution, but it only lives in Drupal. I added an information banner that says to use pega_react_multi_select module. The info is also added by Dan in the comment on [ticket SN-243](https://pegadigitalit.atlassian.net/browse/SN-243)

**Review by remydenton — APPROVED** [review]

**Comment by krlucas** [comment]
@MarcinMr Thanks. Those updates are helpful.

MyPega supports reordering existing projects via drag and drop.
2022-06-28_09-57-27 copy

I don't understand the placement of the drag handle in your mockup. It should go next to each existing project.


cc @colbytcook

**Comment by colbytcook** [comment]
@MarcinMr I am not sure of the mobile behavior of the "My Experience" section, particularly the vertical stacking of the "drag", "content" and the "edit" options. Should maintain a horizontal layout throughout?

File: docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig [hunk 2]
```diff
@@ -0,0 +1,2095 @@
...
+    {% include '@bolt-components-form/form.twig' with {
+      children: form_children
+    } only  %}
+  {% endset %}
+  {% include '@bolt-layouts-layout/layout.twig' with {
+    content: layout_content,
+    padding_top: 'large',
+    padding_bottom: 'large',
+    template: [
+      '75@from-small',
```
Comment (colbytcook) [inline docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig:2084]: @MarcinMr after reviewing the responsive behavior I think we should switch this to '75@from-medium' in order to give the form a little bit more breathing room around ~700 pixels device view size.

**Review by colbytcook — APPROVED** [review]

File: docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig (hunk 2 above)
Comment (colbytcook) [reply docs-site/src/pages/pattern-lab/_patterns/50-pages/85-user/-user-profile-edit.twig]: Updated, closing this comment

**Pull request merged**